src/Str.php

```diff
@@ -662,6 +662,42 @@ public function contains(string $value): bool
         }
     }
 
+    /**
+     * Check if the current string starts with the given string
+     *
+     * @param string $value
+     *
+     * @return bool
+     */
+    public function startsWith(string $value): bool
+    {
+        if ('' === $value) {
+            return false;
+        }
+
+        try {
+            return 0 === $this->position($value);
+        } catch (SubstringException $e) {
+            return false;
+        }
+    }
+
+    /**
+     * Check if the current string ends with the given string
+     *
+     * @param string $value
+     *
+     * @return bool
+     */
+    public function endsWith(string $value): bool
+    {
+        if ('' === $value) {
+            return false;
+        }
+
+        return $value === (string) $this->substring(-strlen($value));
```

Comment:
you can't directly use `strlen` as it doesn't support multibyte, instead you should do: 
```suggestion
        return (string) $this->takeEnd(self::of($value, $this->encoding)->length()) === $value;
```